src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -486,8 +486,8 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	appendStringInfo(buf, "%s ", stmt->is_grant ? " TO " : " FROM ");
 	AppendRoleList(buf, stmt->grantee_roles);
 	AppendGrantWithAdminOption(buf, stmt);
-	AppendGrantRestrictAndCascadeForRoleSpec(buf, stmt->behavior, stmt->is_grant);
 	AppendGrantedByInGrantForRoleSpec(buf, stmt->grantor, stmt->is_grant);
+	AppendGrantRestrictAndCascadeForRoleSpec(buf, stmt->behavior, stmt->is_grant);
 	appendStringInfo(buf, ";");
```

Comment:
Do we really need to change the order of AppendGrantRestrictAndCascadeForRoleSpec? If we need, is this fixing another bug?

Reply:
Yes. This one appeared the tests in here. Restrict should be at the end od the statement

Reply:
Which PR? I merged the https://github.com/citusdata/citus/pull/7451 already

Reply:
I didn't mean #7451. Above change seems like another bug fix because otherwise it wouldn't appear in the diff view as #7451 is merged already.

Can we apply above one liner fix with a simple test in another PR first?